app/src/main/java/org/oppia/app/drawer/NavigationDrawerSwitchProfileDialogFragment.kt

```diff
@@ -0,0 +1,66 @@
+package org.oppia.app.drawer
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.view.Menu
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.view.ContextThemeWrapper
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.fragment.app.DialogFragment
+import org.oppia.app.R
+import org.oppia.app.profile.ProfileChooserActivity
+
+/** [DialogFragment] that gives option to either cancel or exit current profile */
+class NavigationDrawerSwitchProfileDialogFragment : DialogFragment() {
+
+  private lateinit var menu: Menu
```

Comment:
One suggestion I am looking for here:
Another approach is where we can have an interface and rather than passing `menu` and `drawerLayout` object we can implement an interface in `NavigationDrawerFragment` and use it from there. 
The only thing we have to achieve with these objects is 
```
menu.getItem(
          NavigationDrawerItem.HOME.ordinal
        ).isChecked =
          true
        drawerLayout.closeDrawers()
```

Reply:
This approach does sound correct. But PTAL the top level comment of mine.